### PR TITLE
Feature ETP-2583: Modified dependencies from artifacts file

### PR DESCRIPTION
**Pull request by Matias-Bernal**
This pull request refactors and organizes dependency management, build configuration, and setup logic for the project. The main changes include a comprehensive reorganization of the `artifacts.list.COMPILATION.gradle` dependencies, updates to build and pipeline configurations, and the removal of legacy setup logic for Eclipse components.

**Dependency and Build System Improvements**

* Major reorganization and grouping of dependencies in `artifacts.list.COMPILATION.gradle`, including updates, removals, and thematic categorization for improved maintainability. This includes adding new versions, removing duplicates, and grouping by library function (e.g., authentication, reporting, database drivers, logging, etc.). [[1]](diffhunk://#diff-9749a3567917b7ba08e7384f3a7eaa2660928cb3ec8aeb48b17346fe97c666e4R2-L23) [[2]](diffhunk://#diff-9749a3567917b7ba08e7384f3a7eaa2660928cb3ec8aeb48b17346fe97c666e4L34-R80) [[3]](diffhunk://#diff-9749a3567917b7ba08e7384f3a7eaa2660928cb3ec8aeb48b17346fe97c666e4R94-R142)
* Updated Ant build configuration in `build.xml` to use `base.lib.location` instead of `base.lib` for library paths, and removed obsolete references to `ob-rhino-1.6R7.jar`. [[1]](diffhunk://#diff-766797f233c18114f9499750cf1ffbf3829aeea50283850619c01bd173132021L295-R296) [[2]](diffhunk://#diff-766797f233c18114f9499750cf1ffbf3829aeea50283850619c01bd173132021L353-R353)

**Pipeline and Environment Updates**

* Changed the default plugin cloning behavior in both Jenkins pipeline files (`Jenkinsfile` and `JenkinsfileOracle`) to set `CLONE_PLUGIN` to `FALSE`, and updated the Tomcat version to `9.0.111` for both Postgres and Oracle pipelines. [[1]](diffhunk://#diff-cbcae929c37dcfbdd460599044fdea73b07fb439fe8389000ba87d682db55fb6L25-R25) [[2]](diffhunk://#diff-cbcae929c37dcfbdd460599044fdea73b07fb439fe8389000ba87d682db55fb6L37-R37) [[3]](diffhunk://#diff-cffd5217f94e2fca7af2f8f9340f5fcb95c2175ba8f63e8cb6a44f7c3ccb46deL23-R23) [[4]](diffhunk://#diff-cffd5217f94e2fca7af2f8f9340f5fcb95c2175ba8f63e8cb6a44f7c3ccb46deL34-R34)

**Codebase Cleanup**

* Removed all logic related to the Eclipse `.settings/org.eclipse.wst.common.component` file from `ConfigurationApp.java`, including its setup and property replacement methods, as well as references in the setup workflow. [[1]](diffhunk://#diff-3c0d155eb8042aa18c7efb2ee3ad63b865244f23f3fc75245371a8ece8c7ade8L68) [[2]](diffhunk://#diff-3c0d155eb8042aa18c7efb2ee3ad63b865244f23f3fc75245371a8ece8c7ade8L178) [[3]](diffhunk://#diff-3c0d155eb8042aa18c7efb2ee3ad63b865244f23f3fc75245371a8ece8c7ade8L227-L228) [[4]](diffhunk://#diff-3c0d155eb8042aa18c7efb2ee3ad63b865244f23f3fc75245371a8ece8c7ade8L255-L275)
* Stopped copying the `.classpath` template file in the setup process, likely reflecting a move away from Eclipse-specific configuration.

**Documentation**

* Removed the outdated note in `lib/README` that referenced updating the Openbravo libraries wiki page.

### CODE REVIEW

File: artifacts.list.COMPILATION.gradle [hunk 1]
```diff
@@ -1,101 +1,144 @@
 List<String> _dependenciesListCOMPILATION = [
```
Comment (etendobot) [inline artifacts.list.COMPILATION.gradle]: **Suggestion**

For maximum compatibility, use plain comments without emojis or special characters.

```groovy
// Etendo Core and Internal Components
```

**Rationale**: Configuration and build files should remain as portable and environment-agnostic as possible. Emojis could interfere with parsing under some CI/automation tools.

File: build.gradle [hunk 2]
```diff
@@ -11,6 +11,7 @@ plugins {
 }
 compileJava.options.encoding = "UTF-8"
 apply from: "artifacts.list.COMPILATION.gradle"
+apply from: "artifacts.list.TEST.gradle"
```
Comment (etendobot) [inline build.gradle]: **Suggestion**

Consider documenting the purpose of the new artifacts list for tests at the top of the build file for future maintainers.

```groovy
apply from: "artifacts.list.TEST.gradle"
```

**Rationale**: Including a brief comment when adding new artifact sources helps developers quickly understand the reason for the change and aligns with Etendo documentation best practices.

File: artifacts.list.COMPILATION.gradle [hunk 3]
```diff
@@ -1,101 +1,141 @@
...
+
+  // XML Binding and StAX Processing
+  'org.glassfish.jaxb:jaxb-runtime:2.3.1',
+  'org.glassfish.jaxb:txw2:2.3.1',
+  'org.jvnet.staxex:stax-ex:1.8',
+
+  // Template and Scripting Engines
+  'org.freemarker:freemarker:2.3.16',
+  'org.mozilla:rhino:1.8.0',
+  'org.mozilla:rhino-engine:1.8.0',
```
Comment (etendobot) [inline artifacts.list.COMPILATION.gradle]: **Suggestion**

Review the impact of updating the Rhino engine from 1.7.13 to 1.8.0 as Rhino is used for JavaScript support in Etendo custom scripting. Validate compatibility with existing scripts and modules that may rely on the previous version.

```java
'org.mozilla:rhino:1.8.0',
```

**Rationale**: Major version or implementation changes can introduce subtle backward compatibility issues or behavioral changes in embedded scripting. Etendo modules that rely on Rhino should be tested to confirm continued compatibility.

**Review by etendobot — APPROVED** [review]

File: artifacts.list.COMPILATION.gradle [hunk 4]
```diff
@@ -1,101 +1,145 @@
...
```
Comment (etendobot) [inline artifacts.list.COMPILATION.gradle]: **Suggestion**

Upgrading 'org.mozilla:rhino' from 1.7.13 to 1.8.0 may introduce behavioral/API changes or compatibility issues. Verify that downstream code remains compatible and tests pass after bump.

```groovy
'org.mozilla:rhino:1.8.0',
```

**Rationale**: Major version upgrades—especially in core components or scripting engines—can introduce breaking changes or subtle behavior shifts.

File: artifacts.list.COMPILATION.gradle [hunk 5]
```diff
@@ -1,101 +1,145 @@
...
+  // Reporting and Document Generation
+  'com.lowagie:itext:2.1.7',
+  'jfree:jfreechart:1.0.12',
+  'net.sf.jasperreports:jasperreports:6.17.0',
+  'net.sf.jasperreports:jasperreports-fonts:6.17.0',
+
+  // Database Drivers and Persistence Connectors
+  'com.oracle.database.jdbc:ojdbc8:23.9.0.25.07',
+  'org.postgresql:postgresql:42.7.8',
+
```
Comment (etendobot) [inline artifacts.list.COMPILATION.gradle]: **Suggestion**

PostgreSQL JDBC version change from 42.5.4 to 42.7.8 – verify DB compatibility across supported PG versions and test edge cases for datatype handling.

```groovy
'org.postgresql:postgresql:42.7.8',
```

**Rationale**: Core driver updates can impact DB connectivity and ORM mapping, particularly if Etendo modules rely on Postgres-specific features or quirks.

File: artifacts.list.COMPILATION.gradle [hunk 6]
```diff
@@ -1,101 +1,145 @@
...
-  'com.fasterxml.jackson.core:jackson-annotations:2.11.2', 
-  'com.fasterxml.jackson.core:jackson-core:2.11.2', 
-  'com.fasterxml.jackson.core:jackson-databind:2.11.2', 
-  'com.fasterxml:classmate:1.3.4', 
+  'com.etendoerp:font.bitstreamverasans:1.0.0',
+  'com.etendoerp:YUIAnt:1.0.0',
+  'com.etendoerp:yuicompressor:2.4.2',
+
+  // JSON and Serialization Libraries
+  'com.fasterxml.jackson.core:jackson-annotations:2.11.2',
```
Comment (etendobot) [inline artifacts.list.COMPILATION.gradle]: **Suggestion**

Double-check if 'org.codehaus.jettison:jettison:1.3' should be 'com.etendoerp:jettison:1.3'. In prior Etendo artifact lists, the groupId often used is 'com.etendoerp' to ensure internal artifact resolution and shadowing. Mixing sources can cause conflicts.

```plaintext
+  // JSON and Serialization Libraries
+  'com.fasterxml.jackson.core:jackson-annotations:2.11.2',
+  'com.fasterxml.jackson.core:jackson-core:2.11.2',
+  'com.fasterxml.jackson.core:jackson-databind:2.11.2',
+  'com.fasterxml:classmate:1.3.4',
+  'com.etendoerp:jettison:1.3',
```

**Rationale**: Consistency in artifact group IDs is important for builds, especially when Etendo republishes or patches external dependencies. Using a different group may result in using an unsupported or incompatible jar.
